### PR TITLE
POC of automated Storybook accessibility/interaction testing  

### PR DESCRIPTION
**Pull request by LFDanLu**
Pros:
- can replay the interaction test step by step
- catches more accessibility issues since it runs in browser 

Cons:
- needs storybook running 
- reporting from the automated test is kinda hard to read
- some finicky behavior from interaction panel

TODO: followup
- figure out if can run accessibility tests with different themes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

1. Try running yarn test-storybook
2. Try going to the "With state" HelpText story and the "Default"/"Sections"/"Keyboard tab focus" Picker story and play around with the interaction addon panel.

## 🧢 Your Project:

RSP

### CODE REVIEW

**Comment by LFDanLu** [comment]
This PR will need to wait for all of our stories to be converted to CSF 3, for now will serve as an example of additional storybook accessiblity/in browser interaction testing + example of using axe-core in our tests

**Comment by LFDanLu** [comment]
Running the accessibility test yields the following results:
```
Test Suites: 32 failed, 84 passed, 116 total
Tests:       202 failed, 959 passed, 1161 total
```
Spot checked the failures against the accessibility addon in storybook and looked to be about the same. Will go through and try to hide the ones that we know are false positives

File: package.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
     "build:docs": "DOCS_ENV=staging parcel build 'packages/@react-{spectrum,aria,stately}/*/docs/*.mdx' 'packages/react-aria-components/docs/*.mdx' 'packages/@internationalized/*/docs/*.mdx' 'packages/dev/docs/pages/**/*.mdx'",
     "test": "yarn jest",
     "test-strict": "cross-env STRICT_MODE=1 yarn jest",
+    "test-storybook": "test-storybook --url http://localhost:9003  --browsers chromium --no-cache",
```
Comment (reidbarber) [inline package.json:24]: Could this start storybook first (if not running), then run the test runner? Seems like we could, but also I don't see anything in their docs about it.

**Review by reidbarber — COMMENTED** [review]
Awesome!

Have you looked into the chromatic interactions integration? Maybe that's something we could use: https://www.chromatic.com/docs/interactions

File: package.json (hunk 1 above)
Comment (LFDanLu) [reply package.json:24]: I believe we could do something like https://github.com/adobe/react-spectrum/blob/ba727bdc0c4a57626131e84d9c9b661d0b65b754/scripts/verdaccio.sh#L45-L48 where we wait until storybook outputs "Built in" and then run `test-storybook`.

**Comment by LFDanLu** [comment]
@reidbarber 

> Have you looked into the chromatic interactions integration? Maybe that's something we could use: https://www.chromatic.com/docs/interactions

Didn't realize that chromatic supported interactions, thanks for the link! Looks like we'd just have to write play functions in our chromatic stories and they will be run and snap shotted by Chromatic automatically. Def something we can explore using if we have a use case, but I'd like to get people's opinion of writing interaction tests via Storybook first (aka is it worth it or do we prefer the current react testing library tests?) IMO there are some use cases that are easier to test in browser, especially for cases that would require mocking of things like clientHeight/raf/etc

**Comment by reidbarber** [comment]
>IMO there are some use cases that are easier to test in browser, especially for cases that would require mocking of things like clientHeight/raf/etc

@LFDanLu I like this idea a lot. I think it is better coverage and better DX. Makes sense for any interactions we have that come up similar to DnD and table column resizing.

**Pull request closed without merging**